contracts/integrations/custom/CustomIntegrationIdleFinance.sol

```diff
@@ -0,0 +1,338 @@
+     * @return uint256                   Call value
+     * @return bytes                     Trade calldata
+     */
+    function _getPreActionCallData(
+        address, /* _strategy */
+        address, /* _asset */
+        uint256, /* _amount */
+        uint256 /* _customOp */
+    )
+        internal
+        pure
+        override
+        returns (
+            address,
+            uint256,
+            bytes memory
+        )
+    {
+        return (address(0), 0, bytes(''));
+    }
+
+    /**
+     * (OPTIONAL) Return post action calldata
+     *
+     * hparam  _strategy                 Address of the strategy
+     * hparam  _asset                    Address param
+     * hparam  _amount                   Amount
+     * hparam  _customOp                 Type of op
+     *
+     * @return address                   Target contract address
+     * @return uint256                   Call value
+     * @return bytes                     Trade calldata
+     */
+    function _getPostActionCallData(
+        address, /* _strategy */
+        address, /* _asset */
+        uint256, /* _amount */
+        uint256 /* _customOp */
+    )
+        internal
+        pure
+        override
+        returns (
+            address,
+            uint256,
+            bytes memory
+        )
+    {
+        return (address(0), 0, bytes(''));
+    }
+
+    /**
+     * (OPTIONAL). Whether or not the pre action needs an approval.
+     * Only makes sense if _getPreActionCallData is filled.
+     *
+     * hparam  _asset                     Asset passed as param
+     * hparam  _tokenDenominator          0 for enter, 1 for exit
+     * @return address                    Address of the asset to approve
+     * @return address                    Address to approve
+     */
+    function _preActionNeedsApproval(
+        address, /* _asset */
+        uint8 /* _customOp */
+    ) internal pure override returns (address, address) {
+        return (address(0), address(0));
+    }
+
+    /**
+     * (OPTIONAL). Whether or not the post action needs an approval
+     * Only makes sense if _getPostActionCallData is filled.
+     *
+     * hparam  _asset                     Asset passed as param
+     * hparam  _tokenDenominator          0 for enter, 1 for exit
+     * @return address                    Address of the asset to approve
+     * @return address                    Address to approve
+     */
+    function _postActionNeedsApproval(
+        address, /* _asset */
+        uint8 /* _customOp */
+    ) internal pure override returns (address, address) {
+        return (address(0), address(0));
+    }
+
+    function _isValidIdleVault(
+        address _vaultAddress
+    ) internal view returns (bool) {
+      // check for an Idle vault by seeing if it responds to the token() method with a token address
+      (bool success, bytes memory data) = address(_vaultAddress).staticcall{gas: 10000}(abi.encodeWithSignature('token()'));
```

Comment:
Dunno what y'all think about this, or if you have a pattern for this type of stuff already? AFAIK there is no "vault registry" for Idle like there is for Yearn.

Reply:
I would use the price helper for this 

https://etherscan.io/address/0x04Ce60ed10F6D2CfF3AA015fc7b950D13c113be5

Reply:
isn't this going to raise an exception if we pass an invalid token? i assumed that we are trying to avoid that?

Reply:
or wait, you're saying to keep the low level call, but call the price helper? 👌

Reply:
or solidity try...catch works too, i suppose?